source/extensions/filters/network/postgres_proxy/postgres_decoder.cc

```diff
@@ -241,16 +254,26 @@ bool DecoderImpl::onData(Buffer::Instance& data, bool frontend) {
     }
   }
 
-  std::vector<MsgAction>& actions = std::get<1>(msg.get());
+  std::vector<MsgAction>& actions = std::get<2>(msg.get());
   for (const auto& action : actions) {
     action(this);
   }
 
   ENVOY_LOG(debug, "({}) command = {} ({})", msg_processor.direction_, command_,
             std::get<0>(msg.get()));
-  ENVOY_LOG(debug, "({}) length = {}", msg_processor.direction_, getMessageLength());
-  ENVOY_LOG(debug, "({}) message = {}", msg_processor.direction_, getMessage());
+  ENVOY_LOG(debug, "({}) length = {}", msg_processor.direction_, message_len_);
```

Comment:
What about move this two ENVOY_LOG to inside the branch below?

Reply:
Actually ENVOY_LOG is defined exactly in the same fashion: it checks the current logging level and logs only when logging level allows for logging. However, it only works when expression is included in ENVOY_LOG macro. Below (inside the branch) message is parsed outside of ENVOY_LOG macro and would be executed regardless of logging level.
https://github.com/envoyproxy/envoy/blob/master/source/common/common/logger.h#L344-L353

Reply:
WDYT about factoring lines 270-279 into a helper method, drop the if, and then you can simply put
```
ENVOY_LOG(debug, "({}) message = {}", msg_processor.direction_, genDebugMessage());
```
inline with the above logs. That way you don't repeat the level-filtering done in the macro.